central/policy/service/validator.go

```diff
@@ -87,6 +89,9 @@ func (s *policyValidator) internalValidate(policy *storage.Policy, additionalVal
 	errorList.AddError(s.validateExclusions(policy))
 	errorList.AddError(s.validateCapabilities(policy))
 	errorList.AddError(s.validateEventSource(policy))
+	if features.NetworkPolicySystemPolicy.Enabled() {
+		errorList.AddError(s.validateNetworkPolicyEnforcement(policy))
```

Comment:
Interesting that we don't have other policies that aren't enforced.

This is more of a matter of personal preference, but I'd be more inclined to do this in a generic way to make it easier to add not enforceable policies in the future. I was hoping that making a policy not enforceable would be as simple as adding a new field name to a map, for example. 

We could achieve this by creating such a map of `notEnforceablePolicies`, and use as:

```go
// (without feature flag check)
errorList.AddError(s.validateEnforcement(policy))
```

```go
func (s *policyValidator) validateEnforcement(policy *storage.Policy) error {
        if len(policy.GetEnforcementActions()) > 0 {
		for _, s := range policy.GetPolicySections() {
			for _, g := range s.GetPolicyGroups() {
				if _, ok := s.nonEnforceablePolicies[g.GetFieldName()]; ok {
					return errors.Errorf("enforcement of %s field is not allowed", g.GetFieldName())
				}
			}
		}
	}
	return nil
}
```

Reply:
Done. For now I'm hardcoding the values of the `nonEnforceablePolicyFields` map in the `policyValidator`. I'm going to take another a look and see there's a way to somehow *mark* a policy field as not enforceable but I haven't seen it so far.

Reply:
I think it's fine the way you did. There is no policy field structure, there are a lot of different structures linked to each other for each part (field name, metadata, violation message, and so on...). The field name stored in the actual policy is just a generic structure with a name and a list of values.